homeassistant/components/tvoverlay/notify.py

```diff
@@ -0,0 +1,276 @@
+            translation_placeholders={
+                "url": url,
+                "allow_url_list": allow_url_list,
+                "help_url": help_url,
+            },
+        )
+
+    async def _is_valid_file(self, filename: str) -> bool:
+        """Check if a file exists on disk and is in allowlist_external_dirs."""
+
+        file_path = Path(filename).parent
+        if not os.path.exists(file_path) or not os.path.isfile(filename):
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="invalid_file",
+                translation_placeholders={
+                    "filename": filename,
+                },
+            )
+
+        if self.hass.config.is_allowed_path(filename):
+            return True
+
+        allow_file_list = "allowlist_external_dirs"
+        file_name = os.path.basename(filename)
+        help_url = "https://www.home-assistant.io/docs/configuration/basic/"
+        raise ServiceValidationError(
+            translation_domain=DOMAIN,
+            translation_key="remote_file_not_allowed",
+            translation_placeholders={
+                "allow_file_list": allow_file_list,
+                "file_path": str(file_path),
+                "file_name": file_name,
+                "help_url": help_url,
+            },
+        )
+
+    async def _validate_image(self, image_data: str) -> str | None:
+        """Validate image_data is valid and in allowed list."""
+        if image_data.startswith("mdi:"):
+            return image_data
+        if image_data.startswith("http") and await self._is_valid_url(image_data):
+            return image_data
+        if "." in image_data and await self._is_valid_file(image_data):
+            return image_data
+        return None
+
+    async def async_send_message(self, message: str, **kwargs: Any) -> None:
+        """Send a message to a TvOverlay device."""
+        title = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
+        data = kwargs.get(ATTR_DATA) or {}
+        _LOGGER.debug("Notification additional data attributes: %s", data)
+        message_id = data.get(ATTR_ID, str(uuid.uuid1()))
+        app_title = data.get(ATTR_APP_TITLE, DEFAULT_APP_NAME)
+        source_name = data.get(ATTR_SOURCE_NAME, DEFAULT_SOURCE_NAME)
+        app_icon: str | ImageUrlSource | None = None
+        app_icon_data = data.get(ATTR_APP_ICON)
+        if isinstance(app_icon_data, str):
+            app_icon = await self._validate_image(app_icon_data)
+        else:
+            app_icon = (
+                await self._populate_image(app_icon_data) if app_icon_data else None
+            )
+
+        badge_icon = data.get(ATTR_BADGE_ICON, DEFAULT_SMALL_ICON)
+        badge_color = data.get(ATTR_BADGE_COLOR, COLOR_GREEN)
+        position = data.get(ATTR_POSITION, Positions.TOP_RIGHT.value)
+        positions_values = [member.value for member in Positions]
+
+        if position not in positions_values:
```

Comment:
Either move calculating `positions_values` to file level, or rewrite to something like:
```suggestion
        if not any(position == member.value for memeber in Positions):
```